src/components/meals/Meals.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import styles from '../../styles';
+import {
+  MealCategories,
+  MealInput,
+  MealList,
+  MealsComponentProps,
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  FlatList,
+} from '../../models/meals.model';
+
+const MAX_ITEMS = 3;
+
+const Meals: React.FC<MealsComponentProps> = ({ mealCategories }) => {
+  const [mealInput, setMealInput] = useState<Record<MealCategories, MealInput>>({
+    [MealCategories.BREAKFAST]: '',
+    [MealCategories.LUNCH]: '',
+    [MealCategories.DINNER]: '',
+  });
+  const [mealList, setMealList] = useState<Record<MealCategories, MealList>>({
+    [MealCategories.BREAKFAST]: [],
+    [MealCategories.LUNCH]: [],
+    [MealCategories.DINNER]: [],
+  });
+  const [enteredText, setEnteredText] = useState<Record<MealCategories, MealInput>>({
+    [MealCategories.BREAKFAST]: '',
+    [MealCategories.LUNCH]: '',
+    [MealCategories.DINNER]: '',
+  });
+
+  const onMealInputChange = (text: MealInput, category: MealCategories) => {
+    setMealInput((prev) => ({
+      ...prev,
+      [category]: text,
+    }));
+  };
+
+  const handleMealInputPress = (category: MealCategories) => {
+    const currentList = [...mealList[category], mealInput[category]];
+
+    if (currentList.length <= MAX_ITEMS) {
+      setEnteredText((prev) => ({
+        ...prev,
+        [category]: currentList,
+      }));
+
+      setMealList((prev) => ({
+        ...prev,
+        [category]: currentList,
+      }));
+    }
+
+    setMealInput((prev) => ({
+      ...prev,
+      [category]: '',
+    }));
+  };
+
+  return (
+    <View>
+      {mealCategories.map((category, index) => (
+        <View key={index}>
+          <Text style={styles.label}>{category}</Text>
+          <View style={styles.inputContainer}>
+            <TextInput
+              style={styles.input}
+              value={mealInput[category]}
+              placeholder={`What are you having for ${category.toLowerCase()}?`}
+              onChangeText={(text) => onMealInputChange(text, category)}
+            />
+            <TouchableOpacity
+              onPress={() => handleMealInputPress(category)}
+              testID={`plus-button-${category}`}
+            >
+              <Text style={styles.enterButton}>+</Text>
+            </TouchableOpacity>
+          </View>
+          {enteredText[category].length > 0 ? (
+            <>
+              <FlatList
+                data={enteredText[category].slice(0, MAX_ITEMS)}
+                keyExtractor={(item, itemIndex) => itemIndex.toString()}
+                horizontal
+                showsHorizontalScrollIndicator={false}
+                renderItem={({ item }) => (
+                  <View style={styles.enteredItem}>
+                    <Text>{item}</Text>
+                  </View>
+                )}
+              />
+              {enteredText[category].length >= MAX_ITEMS && (
```

Comment:
use conditional (ternary) operator instead of &&
<img width="1018" alt="image" src="https://github.com/BinaHovav/MyMeals-Frontend/assets/33668592/ea7981ee-713c-4c1d-991b-8af0abdb45b4">https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
fixed